### DISCUSSION TITLE
First issue

### DISCUSSION THREAD

**Discussion by jsk95** [ashmind/SourceMock#4] [Q&A] [answered]
https://github.com/ashmind/SourceMock/blob/c4016d737fafdaf8a80d86dfb1bfce12c2bcf287/Tests/ArgumentTests.cs#L44-L51

Is the design of the library so that you don't have to specify the shape of the arguments if you don't care?
What happens in the test when the interface has a no param `ParseToInt32`?

**Comment by ashmind** [comment, marked as answer]
If there is a clear overload conflict, you will have to specify `default` explicitly.
I don't have a good constant for `Any` yet, but default is essentially `Any`.

So something like this
```
mock.Setup.ParseToInt32(default).Returns(1); 
```

**Reply by jsk95** [reply]
Hm if it's for testing, why not just enforce strictness?

**Reply by ashmind** [reply]
What do you mean by "strictness"?

**Reply by jsk95** [reply]
Explicit by default

**Reply by ashmind** [reply]
Part of the design direction — not keen on passing arguments you don't care about.
